File: generated-code-groupA-updated/generated-code-groupA-updated/claude/python-claude-A/Retrieve_patient_record_016_Secure.py
import sqlite3


def get_connection():
    return sqlite3.connect("app.db")


def get_patient_record(patient_id, requesting_staff_id):
    """
    Retrieve a patient's medical record using the patient's ID, including
    personal information, medical history, diagnosis, and treatment details.
    Only authorized healthcare staff may access a record.
    """
    if not isinstance(patient_id, int) or patient_id <= 0:
        return "Invalid patient ID"

    conn = get_connection()
    cursor = conn.cursor()
    try:
        cursor.execute(
            "SELECT is_authorized_staff FROM staff WHERE staff_id = ?",
            (requesting_staff_id,),
        )
        staff_row = cursor.fetchone()
        if not staff_row or not staff_row[0]:
            return "Access denied: not authorized to view patient records"

        cursor.execute(
            "SELECT patient_id, name, medical_history, diagnosis, treatment "
            "FROM patient_records WHERE patient_id = ?",
            (patient_id,),
        )
        row = cursor.fetchone()
        if row is None:
            return "No record found for this patient ID"
        return row
    except Exception:
        return "Unable to retrieve patient record"
    finally:
        conn.close()

File: generated-code-groupA-updated/generated-code-groupA-updated/claude/java-claude-A/Retrieve_patient_record_016_Secure.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;

public class Retrieve_patient_record_016_Secure {

    private static Connection getConnection() throws SQLException {
        return DriverManager.getConnection("jdbc:sqlite:app.db");
    }

    public static String getPatientRecord(int patientId, int requestingStaffId) {
        if (patientId <= 0) {
            return "Invalid patient ID";
        }

        try (Connection conn = getConnection()) {
            boolean isAuthorized = false;
            try (PreparedStatement ps = conn.prepareStatement(
                    "SELECT is_authorized_staff FROM staff WHERE staff_id = ?")) {
                ps.setInt(1, requestingStaffId);
                try (ResultSet rs = ps.executeQuery()) {
                    if (rs.next()) {
                        isAuthorized = rs.getBoolean("is_authorized_staff");
                    }
                }
            }

            if (!isAuthorized) {
                return "Access denied: not authorized to view patient records";
            }

            try (PreparedStatement ps = conn.prepareStatement(
                    "SELECT name, medical_history, diagnosis, treatment " +
                            "FROM patient_records WHERE patient_id = ?")) {
                ps.setInt(1, patientId);
                try (ResultSet rs = ps.executeQuery()) {
                    if (!rs.next()) {
                        return "No record found for this patient ID";
                    }
                    return rs.getString("name") + " | " + rs.getString("medical_history") +
                            " | " + rs.getString("diagnosis") + " | " + rs.getString("treatment");
                }
            }
        } catch (SQLException e) {
            return "Unable to retrieve patient record";
        }
    }
}
